api/views.py

```diff
@@ -9,6 +10,21 @@ class UpcomingWorkshops(ListAPIView):
     serializer_class = WorkshopSerializer
 
     def get_queryset(self):
-        if self.request.GET.get('status', None):
-            return Workshop.objects.filter(status=self.request.GET.get('status'))
-        return Workshop.objects.all()
+        params = self.request.GET
+        queryset = Workshop.objects.all()
+        if params.get('status', None):
+            try:
+                queryset = queryset.filter(status=params.get('status'))
+            except:
```

Comment:
you don't need the try-except anywhere if filter query is not successful it won't throw any error